pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -245,9 +306,149 @@ func (r *ReconcileNodeMaintenance) stopNodeMaintenance(nodeName string) error {
 		return err
 	}
 
+	if err := r.invalidateLease(nodeName); err != nil {
+		return err
+    }
+
 	return nil
 }
 
+func (r *ReconcileNodeMaintenance) invalidateLease(nodeName string) error {
+		if r.isLeaseSupported {
+				log.Info("Lease object supported,  invalidating lease")
+
+				nName := apitypes.NamespacedName{Namespace: LeaseNamespace, Name: nodeName}
+				lease := &coordv1beta1.Lease{}
+
+				if err := r.client.Get(context.TODO(), nName, lease); err != nil {
+					return err
```

Comment:
By definition, a lease that doesn't exist is not valid and should not be reported as an error.

Reply:
Does that also apply to stopMaintenanceMode if we see that no lease exists? (that's the only situation when the error return value is checked, all other calls are to remove the lease if transition towards maintenance mode failed.

Reply:
Correct